workbench-core/environments/src/services/environmentTypeConfigService.ts

```diff
@@ -57,31 +40,26 @@ export default class EnvironmentTypeConfigService {
     envTypeId: string,
     envTypeConfigId: string
   ): Promise<EnvironmentTypeConfig> {
-    const response = await this._aws.helpers.ddb
-      .get(this._buildEnvTypeConfigPkSk(envTypeId, envTypeConfigId))
-      .execute();
+    const response = await this._dynamoDbService.get(this._buildEnvTypeConfigPkSk(envTypeConfigId)).execute();
     const item = (response as GetItemCommandOutput).Item;
-    if (item === undefined) {
+    if (item === undefined || (item.dependency as unknown as string) !== envTypeId) {
```

Comment:
why do we need the `as unknown as string` here?

Reply:
dependency is not part of the response , so I didn't really wanted to add another parser for ddbItem, and just calling item.dependency will make compiler complain

Reply:
added getItem method to handle this and avoid using `unknown`